### DISCUSSION TITLE
Regenerating package-lock.json

### DISCUSSION THREAD

**Discussion by morley92** [vercel/turborepo#8581] [Help] [answered]
### Summary

Should it be possible to delete and regenerate a `package-lock.json` file for an existing Turborepo project?

When I'm attempting this, Turborepo is trying to source local packages from our configured npm repository rather than understanding that they are available locally.

```
npm ERR! 404 Not Found - GET https://<some_npm_repository>/@repo%2fui - Package, '@repo/ui', not found.
npm ERR! 404 
npm ERR! 404  '@repo/ui@*' is not in this registry.
```

The [docs suggest](https://turbo.build/repo/docs/crafting-your-repository/structuring-a-repository#package-manager-lockfile) that there is weird behaviour when running Turbo without a Lockfile, but I would have thought `npm install` from a blank slate should work?

Thanks!

**Comment by anthonyshew** [comment]
From what I've seen before, this is a bug with npm as a package manager. It's supposed to be looking up the package in your repository rather than reaching out to the registry. I might be wrong on that, but Turborepo doesn't play a role in package installation or resolution, allowing the package manager to do that.

An alternative syntax you might like more:
```
  "devDependencies": {
    "@repo/ui": "file:@repo/ui",
}
```

From what I understand, this forces npm (the package manager) to only look for those dependencies in the repository, rather than reaching out to the registry.

**Comment by morley92** [comment, marked as answer]
OK my bad! The `packages/ui` folder didn't actually have a `package.json` which was causing this!
